src/test/java/jp/co/yahoo/yosegi/reader/TestYosegiReader.java

```diff
@@ -0,0 +1,307 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.Arguments;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+public class TestYosegiReader {
+
+  private int blockSize = 1024 * 1024 * 4;
+
+  private ColumnBinary createStringPushdownTestColumn() throws IOException{
+    IColumn column = new PrimitiveColumn( ColumnType.STRING , "column" );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 0 );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 1 );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 2 );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 3 );
+
+    IColumnBinaryMaker maker = new UnsafeOptimizeDumpStringColumnBinaryMaker();
+    ColumnBinaryMakerConfig defaultConfig = new ColumnBinaryMakerConfig();
+    ColumnBinaryMakerCustomConfigNode configNode = new ColumnBinaryMakerCustomConfigNode( "root" , defaultConfig );
+    return maker.toBinary( defaultConfig , null , new CompressResultNode() , column );
+  }
+
+  private ColumnBinary createStringPushdownTestColumn2() throws IOException{
+    IColumn column = new PrimitiveColumn( ColumnType.STRING , "column2" );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 0 );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 1 );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 2 );
+    column.add( ColumnType.STRING , new StringObj( "D" ) , 3 );
+
+    IColumnBinaryMaker maker = new UnsafeOptimizeDumpStringColumnBinaryMaker();
+    ColumnBinaryMakerConfig defaultConfig = new ColumnBinaryMakerConfig();
+    ColumnBinaryMakerCustomConfigNode configNode = new ColumnBinaryMakerCustomConfigNode( "root" , defaultConfig );
+    return maker.toBinary( defaultConfig , null , new CompressResultNode() , column );
+  }
+
+  private byte[] createTestBinary() throws IOException {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    Configuration writerConfig = new Configuration();
+    writerConfig.set( "block.size" , Integer.toString( blockSize ) );
+    YosegiWriter writer = new YosegiWriter( out , writerConfig );
+    List<ColumnBinary> list = Arrays.asList( createStringPushdownTestColumn() );
+    // Block-1 Spread-1
+    writer.appendRow( list , 4 );
+    // Block-1 Spread-2
+    writer.appendRow( list , 4 );
+    writer.writeFixedBlock();
+    // Block-2 Spread-1
+    writer.appendRow( list , 4 );
+    // Block-2 Spread-2
+    writer.appendRow( list , 4 );
+    writer.writeFixedBlock();
+    // Block-3 Spread-1
+    writer.appendRow( Arrays.asList( createStringPushdownTestColumn2() ) , 4 );
+
+    writer.close();
+    out.close();
+    return out.toByteArray();
+  }
+
+  @Test
+  public void T_read_1() throws IOException {
```

Comment:
please describe what the code wants to test in the method name instead of number extension.